src/Simulation/CSharpGeneration/RewriteStep.fs

```diff
@@ -61,9 +49,22 @@ type Emitter() =
                 if content <> null then CompilationLoader.GeneratedFile(source, dir, ".dll.g.cs", content) |> ignore
 
             if not compilation.EntryPoints.IsEmpty then
-                let callable = context.allCallables.[Seq.exactlyOne compilation.EntryPoints]
-                let content = EntryPoint.generate context callable
-                CompilationLoader.GeneratedFile(callable.SourceFile, dir, ".EntryPoint.g.cs", content) |> ignore
+
+                let entryPointCallables =
+                    compilation.EntryPoints
+                    |> Seq.map (fun ep -> context.allCallables.[ep])
+
+                let entryPointSources =
+                    entryPointCallables
+                    |> Seq.groupBy (fun ep -> ep.Source.CodeFile)
+
+                let mainSourceFile = "./EntryPoint" |> Path.GetFullPath |> Uri |> CompilationUnitManager.GetFileId
```

Comment:
@bettinaheim Would it matter if it was `"./EntryPoint"` vs `Path.Combine(outputFolder, "EntryPoint")`? Would there be any tricky problems with using the outputFolder as the path?

Reply:
I've gone with the `Path.Combine(outputFolder, "EntryPoint")` idea as I think that will more reliably put the Main file in the root of the output folder, and not accidently in a subfolder.